### PR TITLE
issue 535 center directive naming conflicts, alternative lf-center

### PR DESCRIPTION
**Pull request by nmccready**
feat(lf-center): lfCenter or 'lf-center' added which is a dupe of center. center to be deprecated in 0.9.X . This was still more complicated that I would have preferred.

issue /#535 resolved

### CODE REVIEW

**Comment by nmccready** [comment]
@tombatossals @elesdoar @wylieconlon @simison @thibaudkistler @thibaudkistler and the whole farm please review

File: src/services/leafletHelpers.js [hunk 1]
```diff
@@ -100,7 +100,45 @@ angular.module("leaflet-directive").factory('leafletHelpers', function ($q, $log
         return !_isDefined(value);
     };
 
+    // BEGIN DIRECT PORT FROM AngularJS code base
```
Comment (nmccready) [inline src/services/leafletHelpers.js]: I ended up not using this; so I should possibly remove. But going down the `lf-` path later we possibly might need this.

Let me know if I should kill this.

File: src/directives/center.js [hunk 2]
```diff
@@ -1,193 +1,214 @@
...
-                            //$log.debug("updating center model...", urlCenter);
-                            angular.extend(scope.center, {
-                                lat: urlCenter.lat,
-                                lng: urlCenter.lng,
-                                zoom: urlCenter.zoom
-                            });
-                        }
-                    });
-                }
+centerDirectiveTypes.forEach(function(directiveName) {
```
Comment (nmccready) [inline src/directives/center.js]: using forEach due to jsHint for loop w/ function warning/error

File: test/unit/centerDirectiveSpec.coffee [hunk 3]
```diff
@@ -22,98 +20,100 @@ describe 'Directive: leaflet center', ->
...
-
-        it 'should update the scope.center if leaflet map is moved', (done) ->
-            element = angular.element('<leaflet center="center"></leaflet>')
-            element = $compile(element)(scope)
-
-            @digest scope, ->
-                leafletData.getMap()
-                .then (map) ->
-                    expect(map.getZoom()).toEqual center.zoom
+    for key, directiveName of ['center', 'lf-center']
```
Comment (nmccready) [inline test/unit/centerDirectiveSpec.coffee]: covering the new addition to make sure stuff is the same

File: examples/0101-basic-lf-center-example.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    </script>
+    <style>
+        input {
+            width: 120px;
+            margin-right: 10px;
+        }
+    </style>
+  </head>
+  <body ng-controller="BasicCenterController">
+     <leaflet lf-center="london" width="100%" height="480px"></leaflet>
```
Comment (nmccready) [inline examples/0101-basic-lf-center-example.html]: bu-yaw it works

**Comment by tombatossals** [comment]
@nmccready  that's great work, merged.

**Comment by nimeso** [comment]
I still have this issue when I'm using the google maps directive :( even if I put If-center I still get
angular.js:13550 Error: [$compile:ctreq] Controller 'leaflet', required by directive 'center', can't be found!
From the Google Maps directive.

Any ideas / help would be great. Thanks for all the hard work

**Comment by carlosen14** [comment]
problem still present using leaflet and google maps together

**Pull request merged**